src/nccl_ofi_sendrecv.c

```diff
@@ -819,8 +819,18 @@ static int sendrecv_comm_mr_base_reg(nccl_net_ofi_comm_t *base_comm,
 	key_pool = &device->base.mr_rkey_pool;
 	struct fid_domain *domain;
 	domain = sendrecv_endpoint_get_ofi_domain(ep);
-	ret = sendrecv_mr_base_register(domain, ep->ofi_ep, key_pool,
-					dev_id, ckey, type, &ret_handle);
+	if (mr_cache && nccl_ofi_mr_ckey_is_iovec(ckey)) {
```

Comment:
again, this special case shouldn't exist.

Reply:
Only checking the MR cache now, where the current logic may still have room to improve.